plugins/obs-ffmpeg/obs-ffmpeg-hls-mux.c

```diff
@@ -0,0 +1,339 @@
+#include "obs-ffmpeg-mux.h"
+
+#include <obs-avc.h>
+#include <obs-module.h>
+#include <util/darray.h>
+#include <util/dstr.h>
+#include <util/circlebuf.h>
+#include <util/pipe.h>
+#include <util/threading.h>
+
+#include "obs-ffmpeg-hls-mux.h"
+
+#define do_log(level, format, ...)                       \
+	blog(level, "[ffmpegh hls muxer: '%s'] " format, \
+	     obs_output_get_name(stream->output), ##__VA_ARGS__)
+
+#define warn(format, ...) do_log(LOG_WARNING, format, ##__VA_ARGS__)
+#define info(format, ...) do_log(LOG_INFO, format, ##__VA_ARGS__)
+
+const char *ffmpeg_hls_mux_getname(void *type)
+{
+	UNUSED_PARAMETER(type);
+	return obs_module_text("FFmpegHlsMuxer");
+}
+
+int hls_stream_dropped_frames(void *data)
+{
+	struct ffmpeg_muxer *stream = data;
+	return stream->dropped_frames;
+}
+
+bool hls_deactivate(struct ffmpeg_muxer *stream)
+{
+	if (stream->mux_thread_joinable) {
```

Comment:
I think you need to call deactivate here as well.

Reply:
True!